compaction/compaction_manager.cc

```diff
@@ -998,7 +998,11 @@ class compaction_manager::offstrategy_compaction_task : public compaction_manage
             auto desc = t.get_compaction_strategy().get_reshaping_job(reshape_candidates, t.schema(), iop, sstables::reshape_mode::strict);
             if (desc.sstables.empty()) {
                 // at this moment reshape_candidates contains a set of sstables ready for integration into main set
-                co_await t.update_sstable_lists_on_off_strategy_completion(old_sstables, reshape_candidates);
+                auto desc = sstables::compaction_completion_desc{
+                    .old_sstables = std::move(old_sstables),
+                    .new_sstables = std::move(reshape_candidates)
```

Comment:
You move reshape_candidates here

Reply:
and use it later:

```c++
            // update list of reshape candidates without input but with output added to it
            auto it = boost::remove_if(reshape_candidates, [&] (auto& s) { return input.contains(s); });
```

but there's a break in between.

Reply:
> So it's okay. But infinite loops with breaks are just a way to say "I'm not using goto" while using goto.

agreed. will improve it in a followup.